### ISSUE TITLE
CephFilesystem (and maybe others): No phase in JSON until ready

### ISSUE THREAD

**Issue by dhs-rec** [rook/rook#10767]
When I create a `CephFilesystem` and want to wait for it to finish using
```
kubectl wait -n rook-ceph cephfilesystem.ceph.rook.io/k8sfs --for=jsonpath='{.status.phase}'=Ready --timeout=360s
```
the command fails because there is no ".status.phase" until it's ready (after 60-70 seconds). OTOH, I can do the same for `CephCluster`, like
```
kubectl wait -n rook-ceph cephcluster.ceph.rook.io/rook-ceph --for=condition=Ready --timeout=1500s
```
or
```
kubectl wait -n rook-ceph cephcluster.ceph.rook.io/rook-ceph --for=jsonpath='{.status.phase}'=Ready --timeout=1500s
```
because it provides a value of "Progressing" right from the beginning.

**Deviation from expected behavior:**
Command fails until the state is "Ready"

**Expected behavior:**
Command should allow me to wait for the filesystem to be in state "Ready".
**How to reproduce it (minimal and precise):**
1. Create a filesystem using the `filesystem.yaml` from the examples directory
2. Run the command above to wait for it to be ready

**Comment by dhs-rec** [comment]
1. Yes, it's the `deploy/examples/filesystem.yaml` where only the name has been changed
2. When the `CephFilesystem` becomes ready? Yes
3. For what?

I don't understand how this is related to the `.status.phase` field not being available until it's value is "Ready" (as opposed to `CephCluster`, where it is available right from the beginning.

**Comment by microyahoo** [comment]
check `rook-ceph-operator-57cfdf5c4b-nnglq` log like below through `kubectl logs -n rook-ceph rook-ceph-operator-57cfdf5c4b-nnglq`
```
➜ /root/go/src/deeproute.ai/smd ☞ git:(test4) kubectl get pods -n rook-ceph -o wide
NAME                                           READY   STATUS      RESTARTS       AGE    IP               NODE   NOMINATED NODE   READINESS GATES
rook-ceph-crashcollector-k1-75bdfc7d8-66fj7    1/1     Running     0              55m    10.9.8.72        k1     <none>           <none>
rook-ceph-crashcollector-k2-799d767c5c-4xzlr   1/1     Running     0              55m    10.9.8.73        k2     <none>           <none>
rook-ceph-crashcollector-k3-5ffdf55bc8-5v9jg   1/1     Running     0              55m    10.9.8.74        k3     <none>           <none>
rook-ceph-mgr-a-6b59bd74c9-vb4gk               2/2     Running     2 (145m ago)   164m   10.9.8.74        k3     <none>           <none>
rook-ceph-mgr-b-5b54b58c99-l66qz               2/2     Running     2 (145m ago)   164m   10.9.8.73        k2     <none>           <none>
rook-ceph-mon-a-5dd966dc6c-ggtjf               1/1     Running     1 (145m ago)   165m   10.9.8.73        k2     <none>           <none>
rook-ceph-mon-b-7b4bc774db-wfxlz               1/1     Running     1 (145m ago)   165m   10.9.8.72        k1     <none>           <none>
rook-ceph-mon-c-f5d648d56-sq722                1/1     Running     1 (145m ago)   165m   10.9.8.74        k3     <none>           <none>
rook-ceph-operator-57cfdf5c4b-nnglq            1/1     Running     2 (144m ago)   166m   100.77.170.70    k2     <none>           <none>
rook-ceph-osd-0-75fc6dbbc9-wkxgp               1/1     Running     1 (145m ago)   163m   10.9.8.73        k2     <none>           <none>
rook-ceph-osd-1-79b76bfb49-qwd8f               1/1     Running     1 (145m ago)   163m   10.9.8.72        k1     <none>           <none>
rook-ceph-osd-2-6bdd5f664b-2vn86               1/1     Running     1 (145m ago)   163m   10.9.8.74        k3     <none>           <none>
rook-ceph-osd-3-6b9b754666-596t9               1/1     Running     1 (145m ago)   163m   10.9.8.73        k2     <none>           <none>
rook-ceph-osd-4-664946cdfd-pkds8               1/1     Running     1 (145m ago)   163m   10.9.8.72        k1     <none>           <none>
rook-ceph-osd-5-7ffd95f56b-ds4wv               1/1     Running     1 (145m ago)   163m   10.9.8.74        k3     <none>           <none>
rook-ceph-osd-prepare-k1--1-shrd2              0/1     Completed   0              143m   10.9.8.72        k1     <none>           <none>
rook-ceph-osd-prepare-k2--1-96ls7              0/1     Completed   0              143m   10.9.8.73        k2     <none>           <none>
rook-ceph-osd-prepare-k3--1-2lvkl              0/1     Completed   0              143m   10.9.8.74        k3     <none>           <none>
```

**Comment by dhs-rec** [comment]
Again: What is this info good for?

The problem is with the `cephfilesystem.ceph.rook.io` CRD, which doesn't provide a `.status.phase` field in its JSON output (for example if you do an `kubectl -n rook-ceph describe cephfilesystem.ceph.rook.io/<name> -o json`) until the state is "Ready", which makes it impossible to run above `kubectl wait` command to wait for it. This *would* be possible if the CRD provided this field right away, maybe with a value of "Pending".

**Comment by microyahoo** [comment]
Sorry for misunderstanding what you meant earlier. I noticed that `CephFilesystem.Status` is a pointer to struct, but `CephCluster.Status` is a struct.
https://github.com/rook/rook/blob/6dbf415492d91afdb6ccbea670be375ac6802dc6/pkg/apis/ceph.rook.io/v1/types.go#L53
https://github.com/rook/rook/blob/6dbf415492d91afdb6ccbea670be375ac6802dc6/pkg/apis/ceph.rook.io/v1/types.go#L997

**Comment by dhs-rec** [comment]
Looked at that file. Looks pretty much like AmigaOS code of the late 80s where they simulated OOP in pure C... ;-)

**Comment by dhs-rec** [comment]
Obviously, two months isn't enough for fixing an issue, so please disable that lousy bot or at least give it a longer time period before chiming in.

**Comment by travisn** [comment]
> Obviously, two months isn't enough for fixing an issue, so please disable that lousy bot or at least give it a longer time period before chiming in.

But aren't you glad it reminded us to work on it? :)

**Comment by microyahoo** [comment]
Is it recommended to use a structure or a pointer for resource `status` in k8s? I noticed that there are two ways to use it in rook. @travisn

**Comment by dhs-rec** [comment]
> But aren't you glad it reminded us to work on it? :)

A simple reminder doesn't attach a "wontfix" label to things... ;-)

**Comment by travisn** [comment]
@aruniiird Are you still looking at this?

**Comment by travisn** [comment]
> > @aruniiird Are you still looking at this?
> 
> Yes @travisn. Unfortunately my system got (/home/USER completely) formatted a couple of times during the testing. I finally created a separate/different USER and started testing. Will update here

Oh no! Hopefully you can test in the future in a VM, or minikube where it's isolated.

**Comment by dhs-rec** [comment]
Please remove this bot.

**Comment by travisn** [comment]
Actually this is expected to be resolved in v1.10.11, as #11453 was merged for that release. 
@dhs-rec If you see otherwise, we will certainly reopen it.
